### PR TITLE
add MCC to BinaryClassificationEvaluator

### PR DESCRIPTION
**Pull request by JINO-ROHIT**
This PR links #3027 and adds the Matthews correlation coefficient.

Who can review?
@tomaarsen

### CODE REVIEW

File: sentence_transformers/evaluation/BinaryClassificationEvaluator.py [hunk 1]
```diff
@@ -275,11 +276,15 @@ def compute_metrices(self, model: SentenceTransformer) -> dict[str, dict[str, fl
...
 
+            predicted_labels = (scores >= f1_threshold) if greater_is_better else (scores <= f1_threshold)
+            mcc = matthews_corrcoef(labels, predicted_labels)
+
             logger.info(f"Accuracy with {name}:           {acc * 100:.2f}\t(Threshold: {acc_threshold:.4f})")
             logger.info(f"F1 with {name}:                 {f1 * 100:.2f}\t(Threshold: {f1_threshold:.4f})")
             logger.info(f"Precision with {name}:          {precision * 100:.2f}")
             logger.info(f"Recall with {name}:             {recall * 100:.2f}")
             logger.info(f"Average Precision with {name}:  {ap * 100:.2f}\n")
+            logger.info(f"Matthews Correlation with {name}: {mcc:.4f}\n")
```
Comment (reissaavedra) [inline sentence_transformers/evaluation/BinaryClassificationEvaluator.py:287]: Hi @JINO-ROHIT ,  
I think you could also add this output to the initial documentation of the BinaryClassificationEvaluator class.

Comment (JINO-ROHIT) [reply sentence_transformers/evaluation/BinaryClassificationEvaluator.py:288]: done!

**Review by reissaavedra — APPROVED** [review]

**Comment by JINO-ROHIT** [comment]
@tomaarsen gentle ping to have a look

File: sentence_transformers/evaluation/BinaryClassificationEvaluator.py (hunk 1 above)
Comment (tomaarsen) [reply sentence_transformers/evaluation/BinaryClassificationEvaluator.py]: Good suggestion!

**Pull request merged**